### ISSUE TITLE
Implement writing of page level column indexes in native parquet writer

### ISSUE THREAD

**Issue by raunaqmorarka** [trinodb/trino#9359]
Currently data written through native parquet writer does not contain page level indexes

cc: @martint @findepi

**Comment by sopel39** [comment]
@raunaqmorarka this is still not done, right?

**Comment by raunaqmorarka** [comment]
Yes, this is still pending and worth doing at some point. I'll reproduce my general comment about page index filtering from elsewhere:

In practise I have not found page indexes to improve performance much because page level min/max indexes are not very selective unless the data is sorted. If the data is sorted, then row group pruning already provides most of the benefit.
When multiple columns are read with a selective predicate on one of columns, then the lazy loading of blocks in the engine already allows orc/parquet readers to skip decoding for filtered rows of the remaining columns after a selective predicate is applied to a column.
Though page indexes may eliminate reads from S3 for a subset of the parquet pages in a column chunk, in practise reads from nearby positions in a file are often coalesced by the orc/parquet reader to avoid making multiple small reads from S3, thereby often limiting the benefit to just saving decompression + filter evaluation on eliminated pages of the column with selective filter.
Reading page indexes also incurs additional S3 requests because they are not part of the footer, the footer has just references to them. So on the whole, page indexes may not significantly improve perf even though it sounds like a valuable feature in theory.
For TPC benchmark in particular, we had little to gain from page indexes in hive.

I think for page indexes to make a significant difference, we'd would need to have a situation of large row groups and some clustering of data within the row group which enables the page indexes to be selective.
